File: pe-core/src/main/java/org/saar/core/screen/OffScreen.java
package org.saar.core.screen;

public interface OffScreen extends Screen {

    void resize(int width, int height);

    default void assureSize(int width, int height) {
        if (width != getWidth() || height != getHeight()) {
            resize(width, height);
        }
    }

    default void resizeToMainScreen() {
        final int width = MainScreen.getInstance().getWidth();
        final int height = MainScreen.getInstance().getHeight();
        assureSize(width, height);
    }

    void delete();

}

File: pe-core/src/main/java/org/saar/core/screen/OffScreen.kt
package org.saar.core.screen

interface OffScreen : Screen {

    fun resize(width: Int, height: Int)

    fun delete()
}

fun OffScreen.assureSize(width: Int, height: Int) {
    if (width != this.width || height != this.height) {
        resize(width, height)
    }
}

fun OffScreen.resizeToMainScreen() = assureSize(MainScreen.width, MainScreen.height)
